src/main.rs

```diff
@@ -173,13 +186,13 @@ fn run_program(ints: &mut Vec<u16>) {
             }
             16 => {
                 // write the value from <b> into memory at address <a>
-                write(&mut memory, &mut registers, a, b);
+                write2(&mut memory, &mut registers, a, b);
```

Comment:
I found it quite hard to understand the terminology used when talking about the values/memory/address in the spec

Reply:
I had a lot of difficulty too. In the end, I think I mostly decided that the PC (program counter) would point to instruction arguments (which could be the address of a value or a register). However, when storing a value there was tricky code around how to manipulate an address.